.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 The Sigstore Authors
+# SPDX-License-Identifier: Apache-2.0
+
+name: deploy-terraform
+
+on:
+  push:
+    branches:
+      - "main"
+  workflow_dispatch:
+
+concurrency: deploy
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    if: github.repository == 'sigstore/sigstore-devops-tools'
+
+    permissions:
+      contents: read  # clone the repository contents
+      id-token: write # federates with GCP
+
+    steps:
+    - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+    - uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5.0.1
+      with:
+        go-version: '1.22'
+        check-latest: true
+
+    - uses: google-github-actions/auth@55bd3a7c6e2ae7cf1877fd1ccb9d54c0503c457c # v2.1.2
+      id: auth
+      with:
+        workload_identity_provider: "projects/801815070094/locations/global/workloadIdentityPools/tooling-pool/providers/github-actions-provider"
+        service_account: "github-actions@sigstore-support-tooling.iam.gserviceaccount.com"
+
+    # Attempt to deploy the terraform configuration
+    - uses: hashicorp/setup-terraform@651471c36a6092792c552e8b1bef71e592b462d8 # v2.0.0
+      with:
+        terraform_version: 1.8
+
+    - working-directory: ./iac
+      run: |
+        terraform init
+
+        terraform plan
```

Comment:
should this write the plan to a temporary file and then use that file in the apply command below? there is a race condition possible here.

Reply:
yep